app/src/main/java/com/asfoundation/wallet/repository/BdsBackEndWriter.kt

```diff
@@ -19,23 +20,34 @@ open class BdsBackEndWriter(
         .flatMap { wallet -> service.submitProof(proof, wallet.address) }
   }
 
-  override fun hasWalletPrepared(chainId: Int): Single<ProofSubmissionFeeData> {
+  override fun hasWalletPrepared(chainId: Int,
+                                 packageName: String,
+                                 versionCode: Int): Single<ProofSubmissionFeeData>? {
     if (!isCorrectNetwork(chainId)) {
-      if (isKnownNetwork(chainId)) {
-        return Single.just(
+      return if (isKnownNetwork(chainId)) {
+        Single.just(
             ProofSubmissionFeeData(ProofSubmissionFeeData.RequirementsStatus.WRONG_NETWORK,
                 BigDecimal.ZERO, BigDecimal.ZERO))
       } else {
-        return Single.just(
+        Single.just(
             ProofSubmissionFeeData(ProofSubmissionFeeData.RequirementsStatus.UNKNOWN_NETWORK,
                 BigDecimal.ZERO, BigDecimal.ZERO))
       }
-
     }
+
     return defaultWalletInteract.find()
+        .flatMap {
+          service.getCampaign(it.address,
+              packageName, versionCode)
+        }
         .map {
-          ProofSubmissionFeeData(ProofSubmissionFeeData.RequirementsStatus.READY,
-              BigDecimal.ZERO, BigDecimal.ZERO)
+          if (!isEligible(it)) {
```

Comment:
Shouldn't we move the false condition to the else part and keep the true one in the if?